### DISCUSSION TITLE
Domain names in fail2ban log files?

### DISCUSSION THREAD

**Discussion by jost666** [fail2ban/fail2ban#3595] [General]
I run a large server with NGINX and virtual hosts. Very common setup.
Fail2ban watches several log files generated by NGINX.

However, all our domains log to central log files.

Is there any way to carry the domain name all the way to the fail2ban log file?  

For example - here is a fail2ban logfile entry:

2023-10-05 10:20:02,811 fail2ban.filter         [1273]: INFO    [watchbot404] Found 124.76.54.84 - 2023-10-05 10:20:02

Need something like this - so I can parse reports for individual domains:
2023-10-05 10:20:02,811 fail2ban.filter         [1273]: INFO    [watchbot404] Found 124.76.54.84 - 2023-10-05 10:20:02 - www.mydomain.com 

I dug around, but doesnt seem compatible with the architecture.
Trying to avoid running a fail2ban instance PER domain. Way too much overhead.

**Comment by sebres** [comment]
The `Found` message is not customizable at the moment.

Also it'd expect that a filter would capture domain from log (already possible, but must be configured) as well as that fail ticket would contain that to the time point `Found` message gets logged.

You can try to extend it with a patch like this:
```diff
 				logSys.info(
-					"[%s] Found %s - %s", self.jailName, ip, MyTime.time2str(unixTime)
+					"[%s] Found %s - %s - %s", self.jailName, ip, MyTime.time2str(unixTime), tick.getData("F-DOMAIN")
 				)
```
for https://github.com/fail2ban/fail2ban/blob/8eac5f597aaf0c993f4ba006beeeafa28b18bafe/fail2ban/server/filter.py#L714-L715

and to capture it from log, extend `prefregex` or `failregex` in order to capture a domain from log-message:
```diff
- failregex = ^<ADDR> [^"]*"[A-Z]{3,} [^"]+" [3-5]\d\d
+ failregex = ^<ADDR> [^"]*"[A-Z]{3,} [^"]+" [3-5]\d\d ... domain:"<F-DOMAIN>[^"]*</F-DOMAIN>"
```
(where new RE part needs to match the domain in the message)

However I don't follow why one need this info in fail2ban.log.

Either it belongs rather to something like #2304.

Or one can simply get it with `fail2ban-regex`:
```bash
# messages by IP:
fail2ban-regex -o msg /path/to/nginx-central-log watchbot404 | grep -wF "$ip"
# ticket data by IP:
fail2ban-regex -o '<ip> - <F-DOMAIN>' /path/to/nginx-central-log watchbot404 | grep -wF "$ip"
```
Or from fail2bans sqlite-database (matches from banned ticket).

BTW, note that monitoring of accesslogs is not really recommended - see [wiki :: Best practice](https://github.com/fail2ban/fail2ban/wiki/Best-practice) (especially part "[Reduce parasitic log-traffic](https://github.com/fail2ban/fail2ban/wiki/Best-practice#reduce-parasitic-log-traffic)") for details.
